### ISSUE TITLE
Encountering 'Invalid client credentials' in Concourse 6.0

### ISSUE THREAD

**Issue by markstgodard** [arbourd/concourse-slack-alert-resource#70]
Hi All,

I was using this excellent concourse resource on a Concourse 5.x deployment, but after moving to a Concourse 6.0.0 deployment, the username / password login when using fixed / broke notifications started failing.

I ensured that I have a local user in the concourse deployment, and has access to the team.

I can login using fly login using local user, however when running in CI I get:

```
2020/09/11 14:50:36 error getting last build status: error connecting to Concourse: oauth2: cannot fetch token: 401 Unauthorized
Response: {"error":"invalid_client","error_description":"Invalid client credentials."}
```

I have tried latest tag and tag for when started supporting Concourse v6.0.

I also pulled the code for the project and looked at isolating the call to login using the creds,
and I see where it is failing when doing the oauth Password Credentials call
https://github.com/arbourd/concourse-slack-alert-resource/blob/master/concourse/client.go#L96

I'm not sure if I missed something or if anyone else has encountered this recently?

Note, I could be missing something in the v6 concourse deploy but local user login seems to work fine from the fly CLI

Cheers

**Comment by arbourd** [comment]
Just to confirm, `v0.11.0`, `v0.12.0` and `v0.13.0` aren't working for you? I'm surprised by `v0.11.0` the most because (although I don't use Concourse anymore) people confirmed it worked with 6.0.

Concourse 6.5.0 completely changed how auth is done (again) and it's actually much simpler. Is there any chance you might be looking to upgrade? Right now this resource technically supports all versions of Concourse but it's really hard to understand / fix auth issues in older builds of the product, like 6.0.0.

**Comment by markstgodard** [comment]
Hey @arbourd 👋

Thanks for the response. Yeah I have tried v0.[11,12,13] but same error.

I pulled the repo and seems like when `login` and hitting the `/sky/issuer/token` endpoint using local user, it seems to fail.

It's been a while since I have been in the Concourse codebase, so I'm not sure if this endpoint is legit for local user login.
I did notice that when I login using a "local user" using username / password login using the Web UI, it goes to a different endpoint (but not sure if that is the root cause)

![image](https://user-images.githubusercontent.com/405082/93025679-88296c80-f5c5-11ea-918f-437cedc03e7e.png)

Will eventually be upgrading to 6.5 for this cluster, so yeah might have different results

Cheers

**Comment by arbourd** [comment]
Looks like you're right. In retrospect, `v0.11.0` was released long after 6.1.0 came out which has _its own set of auth changes_. Perhaps `v0.11.0` only works with 6.1.0 and up.

I've based all of the logic on how `fly` communicates (rather than how the frontend does). If we take a look at [how password auth works in 6.0.0](https://github.com/concourse/concourse/blob/v6.0.0/fly/commands/login.go#L159-L176), it looks like `/issuer` hasn't arrived yet, but [did in 6.1.0](https://github.com/concourse/concourse/blob/v6.1.0/fly/commands/login.go#L141-L163). 😅

Can you try `v0.10.0` and see if works? If 6.0.0 uses the same `fly` auth as 5.5.0, we can adjust the version support for 6.0.0 and set the new auth for 6.1.0 - 6.4.0 (6.5.0 is different again but I fixed that issue last week 🤞). If that's the issue I can patch master and release a fix.

**Comment by markstgodard** [comment]
Hey @arbourd 

I tried `v0.10.0` with our Concourse 6.0 and http://gph.is/1L8Uu2C

**Comment by markstgodard** [comment]
![image](https://user-images.githubusercontent.com/405082/93029871-22001200-f5e4-11ea-9ff9-8da78f1abcb6.png)

If you wanted to update the version support for 6.0, that would be great.

I'll still with v0.10.0 tag for now for anything < 6.5

Cheers

**Comment by arbourd** [comment]
Sounds good I'll get that out now. Thanks for taking the time to try it out for me!

**Comment by arbourd** [comment]
Released with `v0.14.0`. Hopefully it works and if it doesn't, please just reopen. Thanks!

**Comment by markstgodard** [comment]
Confirmed `latest` is good, thanks!
